.. change::
    :tags: bug, asyncio
    :tickets: 12338

    Fixed bug where :meth:`_asyncio.AsyncResult.scalar`,
    :meth:`_asyncio.AsyncResult.scalar_one_or_none`, and
    :meth:`_asyncio.AsyncResult.scalar_one` would raise ``AttributeError:
    'AsyncResult' object has no attribute '_source_supports_scalars'``.
